packages/core/src/lib/persisted.ts

```diff
@@ -0,0 +1,91 @@
+import { writable, type Writable } from 'svelte/store';
+import type { Jsonifiable } from 'type-fest';
+
+const stores: Record<string, Writable<unknown> | undefined> = {};
+
+const getStorage = (type: 'local' | 'session') => {
+  return type === 'local' ? localStorage : sessionStorage;
+};
+
+const parse = <T>(value: string): T | null => {
+  try {
+    return JSON.parse(value) as T;
+  } catch {
+    return null;
+  }
+};
+
+const createPersisted = <T extends Jsonifiable>(
+  key: string,
+  initialValue: T | null,
+  storage: ReturnType<typeof getStorage>
+) => {
+  const store = writable<T | null>(initialValue, (set) => {
+    const json = storage.getItem(key);
+
+    // eslint-disable-next-line unicorn/no-negated-condition
+    if (json !== null) {
+      set(parse<T>(json));
+    } else {
+      storage.setItem(key, JSON.stringify(initialValue));
+    }
+
+    const handleStorage = (event: StorageEvent) => {
+      if (event.key === key) {
+        // eslint-disable-next-line unicorn/no-negated-condition
+        if (event.newValue !== null) {
+          set(parse<T>(event.newValue));
+        } else {
+          set(null);
+        }
+      }
+    };
+
+    window.addEventListener('storage', handleStorage);
+
+    return () => window.removeEventListener('storage', handleStorage);
+  });
```

Comment:
Combined with my comment on `parse` above, we could simplify the body of our store to 

```suggestion
  initialValue = parse(storage.getItem(key)) ?? initialValue
  
  const store = writable<T | null>(initialValue, (set) => {
    storage.setItem(key, JSON.stringify(initialValue));

    const handleStorage = (event: StorageEvent) => {
      if (event.key === key) {
        set(parse<T>(event.newValue));
      }
    };

    window.addEventListener('storage', handleStorage);

    return () => window.removeEventListener('storage', handleStorage);
  });
```